python/uv/private/lock.bzl

```diff
@@ -15,22 +15,258 @@
 """A simple macro to lock the requirements.
 """
 
-load("@bazel_skylib//rules:write_file.bzl", "write_file")
-load("//python:py_binary.bzl", "py_binary")
+load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
+load("//python:py_test.bzl", "py_test")
 load("//python/private:bzlmod_enabled.bzl", "BZLMOD_ENABLED")  # buildifier: disable=bzl-visibility
 
 visibility(["//..."])
 
-_REQUIREMENTS_TARGET_COMPATIBLE_WITH = select({
-    "@platforms//os:windows": ["@platforms//:incompatible"],
-    "//conditions:default": [],
-}) if BZLMOD_ENABLED else ["@platforms//:incompatible"]
+_uv_toolchain = Label("//python/uv:uv_toolchain_type")
+_py_toolchain = Label("//python:toolchain_type")
 
-def lock(*, name, srcs, out, upgrade = False, universal = True, args = [], **kwargs):
+_LockInfo = provider(
+    doc = "",
+    fields = {
+        "args": "",
+        "py": "",
+        "srcs": "",
+        "template": "",
+        "uv": "",
+    },
+)
+
+def _lock_impl(ctx):
+    args = ctx.attr.args
+    srcs = ctx.files.srcs
+    existing_output = ctx.files.existing_output
+    output = ctx.outputs.output
```

Comment:
I'd prefer not having a pre-declared output.
* Havin a predeclared output makes it ambiguous about what one should use when consuming the output (":name", or ":output.txt" ?).
* Forcing people to depend on the target lets us return extra providers, which consumers can make use of without having to update users to switch off the predeclared output